drivers: display: sdl: move SDL pixel format configuration to devicetree (#104099)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-4.5.rst

  have been updated. Application code using :c:func:`dmic_configure`, :c:func:`dmic_trigger`, and
  :c:func:`dmic_read` is not impacted.

Display
=======

> * The Kconfig options for SDL display pixel format selection have been deprecated in favour of

Reviewer: ```suggestion
* The Kconfig options ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_*`` for SDL display pixel format selection have been removed in favour of
```

  setting the pixel-format property directly in devicetree. The following options are no longer

>   supported:

Reviewer: ```suggestion
  setting the pixel-format property directly in devicetree
```

  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_ARGB_8888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_XRGB_8888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_RGB_888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_MONO01``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_MONO10``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_RGB_565``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_RGB_565X``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_L_8``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_AL_88``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_BGR_888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_ABGR_8888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_RGBA_8888``
  ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_BGRA_8888``

>   ``CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_I_4``

Reviewer: ```suggestion
```

>   Instead, set the pixel-format property on the sdl display node in a devicetree using the

Reviewer: ```suggestion
  on the SDL pseudo-device node using the
```

Author: done
